### PR TITLE
authors: name variations

### PR DESCRIPTION
**Pull request by Panos512**
* Adds name_variations field to the records.

Signed-off-by: Panos Paparrigopoulos <panos.paparrigopoulos@cern.ch>

### CODE REVIEW

File: inspirehep/modules/authors/receivers.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@before_record_index.connect
+def generate_name_variatons(recid, json, *args, **kwargs):
+    """Populate a json record before indexing it.
+    Adds a field for all the possible variations of an authors name
+    :param recid: The id of the record that is going to be indexed.
+    :param json: The json representation of the record that is going to be
+                 indexed.
+    """
+    from inspirehep.modules.authors.utils import author_tokenize
```
Comment (kaplun) [inline inspirehep/modules/authors/receivers.py]: I think you can move this import outside.

**Comment by kaplun** [comment]
Can you recover the corresponding tests from the master branch (or add some if missing?)

**Comment by Panos512** [comment]
@jacquerie has already ported the corresponding test on his localhost so it will be up soon.

File: inspirehep/modules/authors/receivers.py (hunk 1 above)
Comment (Panos512) [inline inspirehep/modules/authors/receivers.py]: done!

**Comment by kaplun** [comment]
Uh!? So the tests would arrive afterwards?

**Comment by Panos512** [comment]
Yeah because Jacopo is constructing the new tests structure and has already ported some tests from the invenio2 version.

File: tests/unit/authors/authors_test_variations.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
```
Comment (jacquerie) [inline tests/unit/authors/authors_test_variations.py]: Filename should be `tests/unit/authors/test_authors_utils.py`.

File: tests/unit/authors/authors_test_variations.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
```
Comment (jacquerie) [inline tests/unit/authors/authors_test_variations.py]: Please add CERN paragraph.

File: tests/unit/authors/authors_test_variations.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+from inspirehep.modules.authors.utils import author_tokenize
+
+
+def test_name_variations():
+    name = "John Richard Ellis"
```
Comment (jacquerie) [inline tests/unit/authors/authors_test_variations.py]: This sounds a little undertested. "John Richard Ellis" is quite standard, as far as names go. Maybe @kaplun can provide more interesting test cases?

**Comment by jacquerie** [comment]
The main problem of this PR is that it depends on Beard, for which we have no clean installation mechanism now (we need to build the Docker image). So this will have to wait : /

**Comment by kaplun** [comment]
Wait: why it depends on beard?

**Comment by kaplun** [comment]
Ah! For the little: `from beard.utils.strings import asciify` in `authors.utils`? Well that function could be easily factored out.

**Comment by jacquerie** [comment]
> For the little: `from beard.utils.strings import asciify` in `authors.utils`?

Yep. We could make that import local in `bai` with a big `FIXME`, though.

**Comment by kaplun** [comment]
But actually we should really factor out all  the beard-utiliy into a separate package that doesn't have any major dependency... (OK a bit OT here). So for the time being I guess @Panos512 can indeed temporary factor out this specific function in this PR

**Comment by jacquerie** [comment]
This has tests for the function that is called, but could benefit from some unit tests to the receiver. Take some ispiration by @mihaibivol's tests here: https://github.com/inspirehep/inspire-next/blob/19921ac71621a8e1ff89ae114fc6864d0193c811/tests/unit/records/test_records_receivers.py

**Comment by kaplun** [comment]
@jacquerie, @Panos512 what is the status of this PR?

File: inspirehep/modules/authors/utils.py [hunk 5]
```diff
@@ -66,6 +68,9 @@ def bai(name):
         last_name = names[0]
         initials = []
 
+    # FIXME: This should be loaded on the outer level when travis is updated.
+    from beard.utils.strings import asciify
```
Comment (kaplun) [inline inspirehep/modules/authors/utils.py]: @Panos512 can you cut & paste from `beard` the `asciify` function into `utils.py` as in:

```python
try:
    from beard.utils.strings import asciify
except ImportError:
    def asciify(...):
         ....
```

In this way we don't depend necessarily on `beard` but if it's there we use it :-)

Comment (jacquerie) [inline inspirehep/modules/authors/utils.py]: Yeah, this solution is much better than moving the import as I suggested. :+1:

Comment (jacquerie) [inline inspirehep/modules/authors/utils.py]: Actually, why don't you put it into `inspirehep/modules/authors/utils.py`?

**Comment by jacquerie** [comment]
Let's raise this PR from the dead, since I need it for #918. Will close and refile against `master`.

**Pull request closed without merging**